certbot/storage.py

```diff
@@ -144,9 +146,51 @@ def _relevant(option):
     from certbot import renewal
     from certbot.plugins import disco as plugins_disco
     plugins = list(plugins_disco.PluginsRegistry.find_all())
-    return (option in renewal.STR_CONFIG_ITEMS
-            or option in renewal.INT_CONFIG_ITEMS
-            or any(option.startswith(x + "_") for x in plugins))
+    return (option in renewal.STR_CONFIG_ITEMS or
+            option in renewal.INT_CONFIG_ITEMS or
+            any(option.startswith(x + "_") for x in plugins))
+
+
+def _is_cli_default(option, value):
+    """Does option have the default value?
+
+    Look through the CLI parser defaults and see if this option is
+    both present and equal to the specified value. If not, return
+    False.
+
+    :returns: True if option has the default value, otherwise False
+    :rtype: bool
+
+    """
+    # pylint: disable=protected-access
+    for x in cli.helpful_parser.parser._actions:
+        if x.dest == option:
+            if x.default == value:
+                return True
+            else:
+                break
+    return False
+
+
+def _should_store_option(option, value):
+    """Returns True if option should be stored in the renewal config.
+
+    :param str option: name of the option to be considered
+    :param value: the value of option
+
+    :returns: True if option should be stored, otherwise False
+    :rtype: bool
+
+    """
+    # Try to find reasons to store this item in the renewal config.  It
+    # can be stored if it is relevant and (it is the server option,
```

Comment:
Maybe the comment should say why the `server` option is always stored, because this represents a kind of unusual special case.